File: Algorithm/Java/src/main/java/com/yugutou/charpter3_array/level1/SearchInsert.java
package com.yugutou.charpter3_array.level1;

/**
 * leetcode35 寻找插入位置
 */
public class SearchInsert {
    public static void main(String[] args) {
        int[] arr = {1, 3, 5, 6};
        System.out.println(searchInsert(arr,2));
    }

    public static int searchInsert(int[] nums, int target) {
        int n = nums.length;
        int left = 0, right = n - 1, ans = n;
        while (left <= right) {
            int mid = ((right - left) >> 1) + left;
            if (target <= nums[mid]) {
                ans = mid;
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }
        return ans;
    }
}

File: Algorithm/Code_Materials/python_source/charpter3_array/level1/单调数组/search_insert.py
# coding=utf-8
class SearchInsert:
    def searchInsert(self, nums, target):
        left, right = 0, len(nums) - 1

        while left <= right:
            # 中点
            mid = (left + right) // 2
            if nums[mid] == target:
                return mid
            elif target > nums[mid]:
                left = mid + 1
            else:
                right = mid - 1
        return left


if __name__ == '__main__':
    nums = [1, 3, 5, 6]
    target = 6
    searchInsert = SearchInsert()
    print searchInsert.searchInsert(nums, target)
